### PR TITLE
Fails test when user tries to use syscalls not supported by forge tes…

### PR DESCRIPTION
**Pull request by tkumor3**
…t runner

[Raise gentle error when user tries to use syscalls not supported by forge test runner](https://github.com/foundry-rs/starknet-foundry/issues/71)

Example of the error message, after call not supported syscall 
Zrzut ekranu 2023-08-4 o 15 24 34

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/tests/integration/syscalls.rs [hunk 1]
```diff
@@ -142,3 +142,29 @@ fn library_call_syscall() {
...
+        &test.path().unwrap().join("src/lib.cairo"),
+        &Some(test.linked_libraries()),
+        &Default::default(),
+        Some(&Utf8PathBuf::from_path_buf(corelib().to_path_buf()).unwrap()),
+        &test.contracts(corelib().path()).unwrap(),
+        &Utf8PathBuf::from_path_buf(predeployed_contracts().to_path_buf()).unwrap(),
+    )
+    .unwrap();
+
+    assert_failed!(result);
```
Comment (piotmag769) [inline crates/forge/tests/integration/syscalls.rs:169]: We could also assert that the output contains "starknet syscalls cannot be used in tests", we have a macro `assert_case_output_contains!` for that

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 2]
```diff
@@ -282,14 +282,19 @@ fn execute_syscall(
     let mut buffer = MemBuffer::new(vm, system_ptr);
 
     let selector = buffer.next_felt252().unwrap().to_bytes_be();
+
+    if std::str::from_utf8(&selector).unwrap() != "CallContract" {
+        return Err(HintError::CustomHint(Box::from(
+            "starknet syscalls cannot be used in tests".to_string()
+        )));
+    }
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:290]: I think you are right that it is the only case in which it happens, but let's ask @MaksymilianDemitraszek and @cptartur

Comment (cptartur) [reply crates/forge/src/cheatcodes_hint_processor.rs:290]: I think it's only syscalls, but not 100% sure about the internals of cairo code

File: crates/forge/tests/integration/syscalls.rs (hunk 1 above)
Comment (cptartur) [reply crates/forge/tests/integration/syscalls.rs:169]: I'd definitely use that

**Review by piotmag769 — APPROVED** [review]
Let's wait for @MaksymilianDemitraszek to check the previous comment

**Review by piotmag769 — APPROVED** [review]

**Review by war-in — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Pull request merged**